packages/paste-core/components/meter/src/index.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import {type BoxProps, Box} from '@twilio-paste/box';
+import {Label} from '@twilio-paste/label';
+import {Text} from '@twilio-paste/text';
+import type {HTMLPasteProps} from '@twilio-paste/types';
+import {useMeter} from '@twilio-paste/react-spectrum-library';
+
+export interface MeterProps extends HTMLPasteProps<'meter'>, Pick<BoxProps, 'element'> {
+  minValue?: number;
+  maxValue?: number;
+  value?: number;
+  label: string;
+  showValueLabel?: boolean;
+  formatOptions?: Intl.NumberFormatOptions; // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#options
+  valueLabel?: string;
+  /*
+   * The following props don't exist on the react-aria useMeter hook but do exist on the HTML meter element.
+   * They can be added back into the Paste Meter API depending on the spec & designs.
+   *
+   * low?: number;
+   * high?: number;
+   * optimum?: number;
+   */
+}
+
+const Meter = React.forwardRef<HTMLMeterElement, MeterProps>(({element = 'METER', ...props}, ref) => {
+  const {label, showValueLabel = Boolean(label), value = 0, minValue = 0, maxValue = 100} = props;
+  const {meterProps, labelProps} = useMeter(props);
+
+  // Calculate the width of the bar as a percentage
+  const percentage = (value - minValue) / (maxValue - minValue);
+  const barWidth = `${Math.round(percentage * 100)}%`;
+
+  return (
+    <Box as="div" {...meterProps} role="meter" maxWidth="size30" element={`${element}_WRAPPER`}>
+      <Box display="flex" justifyContent="space-between" element={`${element}_LABEL_WRAPPER`}>
+        {label && (
+          // @ts-expect-error Not using Label's required `htmlFor` prop, using `aria-labelledby` instead
+          <Label {...labelProps} as="div" element={`${element}_LABEL`}>
```

Comment:
suggestion/thought: you can copy Slider in that Label is a composition and not a prop, then you wont need the ts-expect-error because you would be using htmlFor.

Reply:
I was under the impression that `htmlFor` was for interactive labelled elements, which is why we were using `aria-labelledby` here. Which would also mean that consumers would have to ts-ignore their own Label components if they were composed and using `aria-labelledby`. @SiTaggart did I understand you correctly about that?

Reply:
Almost, just a minor technicality. Labels and the for attribute are for Form elements, Slider, _technially_ is an `input` element under the hood in the DOM. 

<img width="580" alt="image" src="https://github.com/twilio-labs/paste/assets/368249/6afc22ce-998d-4358-bb9c-939817f2396e">

Meter and Progress bar aren't, they are just divs with roles so a `label` html element can't be used. It has to be a div or span, using the label styles and applying laballedby on the element that holds the role / replaces the `input`

Reply:
Ok, so I think it makes the most sense to keep Label in the Meter package like it is right now rather than as a composition. If it was a composition, consumers would either have to ignore the required `htmlFor` prop or style their own labels.

Reply:
An alternative to that might be wrapping our label and creating the `MeterLabel` component. You get the same compositional API (it's flexible and not a prop), and you get to control the attributes like the element it renders. 

You could even fake the same API as label using `htmlFor` but your wrapped label component does all the "inside baseball" that a consumer probably doesn't need to know about, like the technicality of using HTML labels

Reply:
So to the consumer it's similar

```
<MeterLabel htmlFor={uid}>
  my label text
</MeterLabel>
<Meter id={uid} />
<HelpText>
  my help text
</HelpText>
```

But under the hood you do all the "right" things

Reply:
What about exporting it from the Label package, since Progress Bar (and potentially other future components) will need to use it too?

Calling it htmlFor even when it's aria-labelledby under the hood seems potentially misleading even if in most cases consumers don't need to know (or even understand) the difference.

But also, all of this is to avoid one ts-ignore, right? Is it necessary?

Reply:
I'm less fussed about the ts ignore, but more about what it's like to be a consumer. There's kind of been an underlying principle of the system that when picking up similar components they share a familiar API, so you kind of have to do less docs searching and mental cycles.

They shouldn't have to know they the label is technically different, and I don't think people will spend the time to understand it anyway.

I also don't think the Label package should know about it or how it has to be used by another package. It shouldn't care, it's a Meter and Progress only concern. Each of those exporting it's own label is fine IMO

Reply:
I'm not convinced that the <MeterLabel> API is a familiar API but I'm fine with it... also it will probably be more complicated than just <MeterLabel> because there is also the value label and potentially min and max labels too, those are still open questions from the kickoff. Just to call that out.

I guess the ts-ignores will be present in the package in that case too